### PR TITLE
SAAS-1192: Activity Log

### PR DESCRIPTION
**Pull request by TomasBlazek3PG**
image
image

### CODE REVIEW

**Review by nicolalamacchia — APPROVED** [review]

File: src/core/api/events.service.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { createApi, fetchBaseQuery, BaseQueryFn, FetchArgs } from '@reduxjs/toolkit/query/react';
+import { ENDPOINTS } from 'core/api';
+import { EventsSearchRequest, EventsSearchResponse } from 'core/api/types';
+import { oktaAuth } from 'core';
+import { EditProfileError } from '../../components/MarketingBanner/MarketingBanner.types';
```
Comment (nicolalamacchia) [inline src/core/api/events.service.ts:5]: please use absolute path, `import { EditProfileError } from 'components'` should be enough

File: src/pages/ManageOrganization/ActivityLog/ActivityLog.tsx [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        }))[0] || {
+          field_name: 'timestamp',
+          order: 1,
+        },
+      },
+    });
+  }, [pageSize, pageIndex, timeframe, sortModel, eventSearch, orgId]);
+
+  const handlePageChange = (newPageIndex: number) => setPageIndex(newPageIndex);
+  const handlePageSizeChange = (newPageSize: number) => setPageSize(newPageSize);
```
Comment (nicolalamacchia) [inline src/pages/ManageOrganization/ActivityLog/ActivityLog.tsx:66]: shouldn't this be handled automatically by DataGrid?

File: src/pages/ManageOrganization/ActivityLog/ActivityLog.tsx [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  const handleSelectTimeframe = (e: SelectChangeEvent<TimestampRangeType>) => {
+    setPageIndex(0);
+    setTimeframe(e.target.value as TimestampRangeType);
+  };
+  const handleSortModelChange = (model: GridSortModel) => setSortModel(model);
```
Comment (nicolalamacchia) [inline src/pages/ManageOrganization/ActivityLog/ActivityLog.tsx:71]: why is this needed?

File: src/pages/ManageOrganization/ActivityLog/ActivityLog.tsx [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+  const renderDetailCell = ({ row }: { row: AuditEvent }) => {
+    const content = getDetailsText(row);
+
+    return (
+      <div title={content} className={styles.scrollableCell}>
+        {content}
+      </div>
+    );
+  };
```
Comment (nicolalamacchia) [inline src/pages/ManageOrganization/ActivityLog/ActivityLog.tsx:104]: `render...` function are a code smell, you should create separate components for those

File: src/pages/ManageOrganization/ActivityLog/ActivityLog.tsx [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+  const getDetailsText = (row: AuditEvent) => {
+    if ([EventType.LOG_IN, EventType.LOG_OUT, EventType.CHANGE_PASSWORD].includes(row.action_type)) {
+      return `User ${row.okta_user_id} ${eventTypeMapping[row.action_type]}`;
+    }
+
+    return `User ${row.okta_user_id} ${eventTypeMapping[row.action_type]} ${getListOfAffectedObjects(
+      row.objects,
+    )}`;
+  };
```
Comment (nicolalamacchia) [inline src/pages/ManageOrganization/ActivityLog/ActivityLog.tsx:88]: what do these do?

**Review by nicolalamacchia — CHANGES_REQUESTED** [review]

File: src/pages/ManageOrganization/ActivityLog/ActivityLog.tsx (hunk 5 above)
Comment (TomasBlazek3PG) [reply src/pages/ManageOrganization/ActivityLog/ActivityLog.tsx:88]: This creates human readable details column text like "User xxx updated xxxx,xxxx,xxxx instance"

File: src/pages/ManageOrganization/ActivityLog/ActivityLog.tsx (hunk 3 above)
Comment (TomasBlazek3PG) [reply src/pages/ManageOrganization/ActivityLog/ActivityLog.tsx:71]: This is setting the sort model (which way and which column should be sorted). It is then used in the request to BE.

File: src/pages/ManageOrganization/ActivityLog/ActivityLog.tsx (hunk 2 above)
Comment (TomasBlazek3PG) [reply src/pages/ManageOrganization/ActivityLog/ActivityLog.tsx:66]: We need to know what is the size so we can add this info to the request to BE

Comment (nicolalamacchia) [reply src/pages/ManageOrganization/ActivityLog/ActivityLog.tsx:67]: right, I forgot it's paginated on the BE

**Review by nicolalamacchia — CHANGES_REQUESTED** [review]
Looks good, but tests are missing

**Review by nicolalamacchia — DISMISSED** [review]

**Review by nicolalamacchia — APPROVED** [review]

**Pull request merged**